src/library/trackcollection.cpp

```diff
@@ -143,6 +143,38 @@ void TrackCollection::relocateDirectory(QString oldDir, QString newDir) {
 }
 
 bool TrackCollection::hideTracks(const QList<TrackId>& trackIds) {
+    // Warn if tracks have a playlist membership
+    QSet<int> allPlaylistIds;
+    for (const auto& trackId: trackIds) {
+        QSet<int> playlisIds;
+        m_playlistDao.getPlaylistsTrackIsIn(trackId, &playlisIds);
+        QString playlistNames;
+        for (const auto& playlisId: playlisIds) {
+            if (m_playlistDao.getHiddenType(playlisId) != PlaylistDAO::PLHT_SET_LOG) {
+                allPlaylistIds.insert(playlisId);
+            }
+        }
+    }
+
+    if (allPlaylistIds.count()) {
+        QString playlistNames = "\n\n";
```

Comment:
I would collect the names of all playlists in a QStringList that is formatted when building the string for the message box text. Slightly less efficient, but I prefer clean and expressive code.